CedarJava/src/main/java/com/cedarpolicy/model/schema/Schema.java

```diff
@@ -27,6 +28,10 @@
 public final class Schema {
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
+    static {
```

Comment:
I notice that we now load the library in a few places. I don't know if this ends up slow at all or if it notices it's already been loaded, but if nothing else it might be nice to keep the logic in one place. For example, if we end up doing something more sophisticated like packaging built Cedar libraries in the JAR, it would be nice to only have to replace it once.

Perhaps we could move the JNI definitions to a single package-private class? (Or have a flag somewhere...)

Reply:
Wouldn't such a class get really big? And wouldn't this be a one-time cost only paid at initialization time?

Reply:
It is a one-time cost per class. The only thing I could imagine being slightly annoying is if you say pre-construct an `AuthorizationEngine` when your application starts, but then touch the `Schema` or `Policy` classes for the first time in a request context. If calling `loadLibrary` does end up being expensive, then it wouldn't be ideal for this to happen there.

I think something like this would work actually, rather than having to centralise the `native` declarations:

```java
public final class Schema {
  static { CedarLibrary.init(); }
}

static final class CedarLibrary {
  static { System.loadLibrary(...); }

  private CedarLibrary() {}
  public static void init() {}
}
```

...but you're right, this is probably premature optimisation.

Reply:
Good point that the Schema class won't actually get the static initialization until you call `new Schema()` somewhere..